draft/0006-pat.md

```diff
@@ -0,0 +1,127 @@
+
+1. `grant_type`: REQUIRED. The value of this parameter must be `urn:ietf:params:oauth:grant-type:token-exchange` indicates that a token exchange is being performed.
+2. `resource`: OPTIONAL. The resource indicator, the same as other token requests.
+3. `scope`: OPTIONAL. The requested scopes, the same as other token requests.
+4. `subject_token`: REQUIRED. The user's PAT.
+5. `subject_token_type`: REQUIRED. The type of the security token provided in the `subject_token` parameter. The value of this parameter must be `urn:logto:token-type:personal-access-token`.
+
+For security reasons, the client must authenticate with the token endpoint using the client ID and client secret in the `Authorization` header. Which means the client must be a traditional web application.
+
+And organization resources are not available for PATs.
+
+### Response
+
+If the token exchange request is successful, the tenant's token endpoint returns an access token that represents the identity of the user. The response includes the following parameters in the HTTP response entity-body using the `application/json` format.
+
+1. `access_token`: REQUIRED. The access token that represents the identity of the impersonated user.
+2. `issued_token_type`: REQUIRED. The type of the issued token. The value of this parameter must be `urn:ietf:params:oauth:token-type:access_token`.
+3. `token_type`: REQUIRED. The type of the token. The value of this parameter must be `Bearer`.
+4. `expires_in`: REQUIRED. The lifetime in seconds of the access token.
+5. `scope`: OPTIONAL. The scopes of the access token.
+
+### Example token exchange
+
+```http
+POST /oidc/token HTTP/1.1
+Host: tenant.logto.app
+Content-Type: application/x-www-form-urlencoded
+Authorization: Basic <base64(client-id:client-secret)>
+
+grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Atoken-exchange
+&scope=profile
+&subject_token=pat_W51arOqe7nynW75nWhvYogyc
+&subject_token_type=urn%3Alogto%3Atoken-type%3Apersonal-access-token
+```
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+  "access_token": "eyJhbGci...zg",
+  "issued_token_type": "urn:ietf:params:oauth:token-type:access_token",
+  "token_type": "Bearer",
+  "expires_in": 3600,
+  "scope": "profile"
+}
+```
+
+## Drawbacks
+
+1. PATs may be misused or leaked if not monitored or revoked in a timely manner.
+2. PATs may be used to bypass multi-factor authentication if not properly enforced.
+
+## Alternatives
+
+1. Implement a separate authentication flow for non-interactive clients, which may require additional development effort.
+2. Use long-lived access tokens, which may be less secure and harder to manage.
+
+## Future Possibilities
+
+1. Support for PAT rotation and automatic expiration policies.
```

Comment:
rotation is more like a practice? i mean nothing to do with the implementation side for the foreseeable moment.